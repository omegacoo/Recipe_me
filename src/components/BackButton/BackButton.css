.BackButton {
    display: block;
    position: absolute;
    width: 100%;
    height: 5vh;
    bottom: 0;
    background-color: red;
    color: white;
    border-style: none;
    text-align: center;
}