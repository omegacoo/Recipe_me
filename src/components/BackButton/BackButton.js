import React from 'react';

import './BackButton.css';

export default class BackButton extends React.Component {
    handleClick = () => {
        window.history.back();
    };

    render(){
        return(
            <button 
                className='BackButton'
                onClick={this.handleClick}
            >
                back
            </button>
        );
    };
};