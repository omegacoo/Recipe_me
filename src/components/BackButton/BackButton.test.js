import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import BackButton from './BackButton';

describe('BackButton component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');

        ReactDOM.render(
            <BrowserRouter>
                <BackButton />
            </BrowserRouter>, 
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders UI correctly', () => {
        expect(
        renderer.create(
            <BrowserRouter>
                <BackButton />
            </BrowserRouter>).toJSON()
        ).toMatchSnapshot();
    });
});