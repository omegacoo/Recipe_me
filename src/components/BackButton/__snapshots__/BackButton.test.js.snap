// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton component renders UI correctly 1`] = `
<button
  className="BackButton"
  onClick={[Function]}
>
  back
</button>
`;
